app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -122,7 +122,7 @@ def credentials
     collector.credentials.each do |credential|
       inventory_object = persister.credentials.find_or_build(credential.id.to_s)
       inventory_object.name = credential.name
-      inventory_object.userid = credential.username
+      inventory_object.userid = credential.try(:username) || ""
```

Comment:
I still think `|| ""` is a bad idea, this is assuming that we have an issue with nils before we actually have one.  If username cannot be nil we need to add a validation to the model, not sprinkle these all over the place

Reply:
Where might that be, @agrare ? Even with rubymine, navigating around this code is an adventure.

Reply:
Do you mean where is the model?

Reply:
I agree that we can let the `nil` to go in and fix wherever that may fail.

Reply:
Yeah, where is the model?

Reply:
https://github.com/ManageIQ/manageiq/blob/master/app/models/auth_userid_password.rb

But again we shouldn't force it to be non nil unless we actually need to

Reply:
The validation would not really help anyway, since that would just fail and we would need to do ```credential.try(:username) || ""``` or set default value.

Looking at the BZ, it should be enough to have ```inventory_object.userid = credential.try(:username)```